neurodsp/sim/combined.py

```diff
@@ -161,6 +161,38 @@ def sim_peak_oscillation(sig_ap, fs, freq, bw, height):
     return sig
 
 
+@normalize
+def sim_combined_peak(n_seconds, fs, components):
+    """Simulate a combined signal with an aperiodic component and a peak.
+
+    Parameters
+    ----------
+    n_seconds : float
+        Simulation time, in seconds.
+    fs : float
+        Sampling rate of simulated signal, in Hz.
+    components : dict
+        A dictionary of simulation functions to run, with their desired parameters.
+
+    Returns
+    -------
+    sig : 1d array
+        Simulated combined peak signal.
+    """
+
+    sim_names = list(components.keys())
+    assert len(sim_names) == 2, 'Expected only 2 components.'
+    assert sim_names[1] == 'sim_peak_oscillation', \
```

Comment:
Yeh, this is a good point - I think the slight tension is whether `sim_combined_peak` should mimic the API of `sim_combined` (as current, but then it has extra constraints) or mimic `sim_peak_oscillation`. I think you're right - the usage here is a bit more bespoke, and so accepting a very general dict input but requiring specific things doesn't really make sense (as opposed to updating the API to list the expected things). 

I'll update!

Reply:
Edit: I later realized why I did it like this - mimicking sim_combined allows for writing out groups of simulations in the same way, and then passing them through the either `sim_combined` or `sim_combined_peak` in the same way (even though `sim_combined_peak` has some extra requirements), and I think this is useful - changing this would reduce this consistency